lib/elixir/lib/io.ex

```diff
@@ -230,9 +238,9 @@ defmodule IO do
   """
   @spec getn(chardata | String.Chars.t, pos_integer) :: chardata | nodata
   @spec getn(device, chardata | String.Chars.t) :: chardata | nodata
-  def getn(prompt, count \\ 1)
+  def getn(device, count \\ 1)
 
-  def getn(prompt, count) when is_integer(count) do
+  def getn(prompt, count) when is_integer(count) and count > 0 do
```

Comment:
Maybe we should change the first argument here to `device` as well? Just for consistency, and also we're changing this line anyways so the blame won't be affected.